partition/uboot.go

```diff
@@ -28,51 +28,33 @@ import (
 	"github.com/mvo5/uboot-go/uenv"
 )
 
-const (
-	bootloaderUbootDirReal        = "/boot/uboot"
-	bootloaderUbootConfigFileReal = "uEnv.txt"
-
-	// the real uboot env
-	bootloaderUbootFwEnvFileReal = "uboot.env"
-)
-
-// var to make it testable
-var (
-	atomicWriteFile = helpers.AtomicWriteFile
-)
-
-const bootloaderNameUboot bootloaderName = "u-boot"
-
 type uboot struct {
 }
 
-func bootloaderUbootDir() string {
-	return filepath.Join(dirs.GlobalRootDir, bootloaderUbootDirReal)
-}
-
-func bootloaderUbootConfigFile() string {
-	return filepath.Join(bootloaderUbootDir(), bootloaderUbootConfigFileReal)
-}
-
-func bootloaderUbootFwEnvFile() string {
-	return filepath.Join(bootloaderUbootDir(), bootloaderUbootFwEnvFileReal)
-}
-
 // newUboot create a new Uboot bootloader object
 func newUboot() bootLoader {
-	if !helpers.FileExists(bootloaderUbootConfigFile()) {
+	u := &uboot{}
+	if !helpers.FileExists(u.configFile()) {
 		return nil
 	}
 
-	return &uboot{}
+	return u
+}
+
+func (u *uboot) Dir() string {
+	return filepath.Join(dirs.GlobalRootDir, "/boot/uboot")
```

Comment:
That piece of code also suggested having a single directory, such as `/boot/loader`. What prevents us from having it sooner?